packages/framework/src/Model/Product/TopProduct/TopProductRepository.php

```diff
@@ -56,6 +57,10 @@ public function getOfferedProductsForTopProductsOnDomain($domainId, $pricingGrou
             ->orderBy('tp.position')
             ->setParameter('domainId', $domainId);
 
+        if ($limit !== null) {
```

Comment:
nitpick 🤓 

it's not necessary to surround the setMaxResults in the if statement as setMaxResults accepts null as an argument.